app/javascript/packs/admin/displayAds.jsx

```diff
@@ -0,0 +1,26 @@
+import { h, render } from 'preact';
+import { Tags } from '../../display-ad/tags';
+
+Document.prototype.ready = new Promise((resolve) => {
+  if (document.readyState !== 'loading') {
+    return resolve();
+  }
+  document.addEventListener('DOMContentLoaded', () => resolve());
+  return null;
+});
+
+function saveTags() {}
+
+function loadTagsField() {
+  const displayAdsTargetedTags = document.getElementById(
+    'display-ad-targeted-tags',
+  );
+
+  if (displayAdsTargetedTags) {
+    render(<Tags onInput={saveTags} />, displayAdsTargetedTags);
+  }
+}
+
+document.ready.then(() => {
+  loadTagsField();
+});
```

Comment:
TIL about this approach being used in a few components

**Question:** How does this relate to `InstantClick.on('change' ...`? Is it needed because we have to wait for the CSRF token to exist in the DOM or are there other reasons for this?

My first thought would be that this is needed because we don't use InstantClick in the admin area, but I'm not 100% sure about this assumption I'm making. Asking to understand this better but also came to mind it might be worth extracting it to a utility helper function, maybe? Out of scope of this PR if that's the case though 🙂

Reply:
@fdocr sorry for the delayed response on this, I've been away but I'll look into this asap.

Reply:
In some places, like the articleForm.jsx, we do the following: 

```
document.ready.then(() => {
  loadForm();
  window.InstantClick.on('change', () => {
    if (document.getElementById('article-form')) {
      loadForm();
    }
  });
});
```

The `document.ready` is being defined earlier up by 
```
HTMLDocument.prototype.ready = new Promise((resolve) => {
  if (document.readyState !== 'loading') {
    return resolve();
  }
  document.addEventListener('DOMContentLoaded', () => resolve());
  return null;
});
```

which seems to check the state of the document. If the state is not loading then it resolves the promise, else it waits for the `DOMContentLoaded`  event to get fired and then resolves the promise. 

Now for the Instant Click part, I did some more [reading on the Instant Click events 
](http://instantclick.io/scripts) and here's my thoughts:

The `DOMContentLoaded` event usually fires when the HTML document has been completely parsed, and all deferred scripts have downloaded and executed. However, when we use InstantClick, we no longer have a  `DOMContentLoaded` firing on page changes anymore because we're preloading them. Hence, we have to tweak the way we listen for events. 

If any additional preloading has been done on any parts of the page through InstantClick, we want to hook into "that" change event to execute whatever logic we need to. 

However, on the admin, we don't use Instant Click to preload pages, hence the `InstantClick.on('change' ..` is not needed.

> Asking to understand this better but also came to mind it might be worth extracting it to a utility helper function, maybe?

I think we could possibly extract this piece of code:
```
HTMLDocument.prototype.ready = new Promise((resolve) => {
  if (document.readyState !== 'loading') {
    return resolve();
  }
  document.addEventListener('DOMContentLoaded', () => resolve());
  return null;
});
```
into a utility helper as it occurs in a couple of places and is adding a `DOMContentLoaded` event listener. It's kinda implementing the logic of what something like `jQuery.ready` does.

I'll add a little note for me to take care of this at the end of the project in a subsequent PR.
 
Does this makes sense? There may be some gaps here as I am not super familiar with InstantClick and had to do some research myself too.

Reply:
Thanks for the detailed response @Ridhwana! I just read through it and makes a lot of sense, thanks 🙌🏼